stan/math/prim/mat/err/check_matching_dims.hpp

```diff
@@ -47,6 +46,49 @@ namespace stan {
                        "columns of ", name2, y2.cols());
     }
 
+    /**
+     * Check if the two matrices are of the same size.
+     *
+     * This function checks the runtime sizes and can also check the static
+     * sizes as well. For example, a 4x1 matrix is not the same as a vector
+     * with 4 elements.
+     *
+     * @tparam check_compile Whether to check the static sizes 
+     * @tparam T1 Scalar type of the first matrix
+     * @tparam T2 Scalar type of the second matrix
+     * @tparam R1 Rows specified at compile time of the first matrix
+     * @tparam C1 Columns specified at compile time of the first matrix
+     * @tparam R2 Rows specified at compile time of the second matrix
+     * @tparam C2 Columns specified at compile time of the second matrix
+     *
```

Comment:
You don't need spaces between the blocks of parameter comments, but I'll levae that up to you.  Thanks for commenting everything.

Reply:
If you're ambivalent about it, I'll leave the spaces because I find the comments more readable with the spaces.